debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/CommitScn.java

```diff
@@ -321,25 +337,36 @@ public void setSsn(int ssn) {
             this.ssn = ssn;
         }
 
+        public Set<String> getTxIds() {
+            return txIds;
+        }
+
+        public void setTxIds(Set<String> txIds) {
+            this.txIds = txIds;
+        }
+
         public String getFormattedString() {
-            return commitScn.toString() + ":" + (rsId != null ? rsId : "") + ":" + ssn + ":" + thread;
+            return commitScn.toString() + ":" + (rsId != null ? rsId : "") + ":" + ssn + ":" + thread + ":" + Strings.join("-", txIds);
         }
 
         public static RedoThreadCommitScn valueOf(String value) {
-            final String[] parts = value.split(":");
+            final String[] parts = value.split(":", -1);
             if (parts.length == 1) {
                 // Reading a legacy commit_scn entry that has only the SCN bit
                 // Create the redo thread entry with thread 1.
                 // There is only ever a single redo thread commit entry in this use case.
-                return new RedoThreadCommitScn(1, Scn.valueOf(parts[0]), null, 0);
-            }
-            else if (parts.length == 4) {
+                return new RedoThreadCommitScn(1, Scn.valueOf(parts[0]), null, 0, new HashSet<>());
+            } else if (parts.length == 5) {
```

Comment:
This `else if` should be on a new line to satisfy the check-style requirement.

Additionally, we should add a new section with `parts.length=5` rather than replacing the `parts.length=4` so that the logic is forward compatible.